cmp/grid/columns/Column.js

```diff
@@ -377,14 +378,24 @@ export class Column {
         } else if (this.tooltipElement) {
             ret.tooltipValueGetter = ({value}) => value;
             ret.tooltipComponentFramework = class extends Component {
-                getReactContainerClasses() {return ['xh-grid-tooltip']}
+                getReactContainerClasses() {
+                    const agParams = this.props,
+                        {location} = agParams;
+                    return location === 'header' ?
+                        ['ag-tooltip'] :
+                        ['xh-grid-tooltip'];
+                }
                 render() {
                     const agParams = this.props,
-                        {value, rowIndex, api} = agParams,
-                        record = api.getDisplayedRowAtIndex(rowIndex).data;
+                        {rowIndex, api, colDef, location} = agParams;
+
+                    if (location === 'header') {
+                        return div(colDef.headerTooltip);
+                    }
 
-                    // ag-Grid encodes the value, so we decode it before passing to the renderer
-                    return me.tooltipElement(decodeURIComponent(value), {record, column: me, gridModel, agParams});
+                    const record = api.getDisplayedRowAtIndex(rowIndex).data,
+                        value = record.get(colDef.colId);
```

Comment:
Believe we should use `colDef.field` here instead of `colId` - they are going to be the same in almost all cases, but you can have cases with multiple cols pulling the same record field (e.g. to render the same primary value differently). In that case the dev has been required to set unique colIds which won't map onto record fields.

It's also true that `Column.field` is optional and might not be specified - say for a col where there is no one field you want to map to, but instead you are rendering based on the entire record. 

I thought that was gonna be fine - primary value passed to tooltip renderer can be null - but I see that `record.get()` will throw on an unknown field - part of the value-add with the `get()` method is that it checks that access. For this use-case I would pull from `record.data[colDef.field]` - hopefully that's the last little wrinkly... 🤞

Reply:
Ok, thanks for the tips re record.data. I have applied the suggested change.